### DISCUSSION TITLE
Failed to look up nuget package dotnetPackage

### DISCUSSION THREAD

**Discussion by CengAmat** [renovatebot/renovate#29069] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

I have two private packages on Github. ("SimpleCalculator" and "dotnetPackage")
SimpleCalculator has dependency on dotnetPackage.
Installed renovate in both repo.
When renovate scans SimpleCalculator and creates PR for Configure Renovate, it has warning:
![image](https://github.com/renovatebot/renovate/assets/1048693/51ba6b05-5af6-4b58-8902-b52fce394d65)

and this is my renovate.json file:
```
{
     "$schema": "https://docs.renovatebot.com/renovate-schema.json",
     "extends": [
        "config:recommended"
     ],
     "hostRules": [
      {
        "matchHost": "https://nuget.pkg.github.com/cengamat/",
        "hostType": "nuget",
        "encrypted": {
          "token":  < PAT token that has all necessary read permissions  >
        }
      }
    ],
   "packageRules": [
     {
       "matchManagers": ["nuget"],
       "matchSourceUrls": ["https://nuget.pkg.github.com/cengamat/index.json"]
     }
   ]
  }

```
SimpleCalculator.csproj:
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <RestoreSources>https://api.nuget.org/v3/index.json;https://nuget.pkg.github.com/cengamat/index.json</RestoreSources>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="dotnetPackage" Version="1.0.3" />
  </ItemGroup>

</Project>
```

And this is what I see from renovate debug logs:
```
DEBUG: hostRules: no authentication for api.nuget.org
DEBUG: GET https://api.nuget.org/v3/registration5-gz-semver2/dotnetpackage/index.json = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=195)
DEBUG: Datasource 404
{
  "datasource": "nuget"
  "packageName": "dotnetPackage"
  "url": "https://api.nuget.org/v3/registration5-gz-semver2/dotnetpackage/index.json"
}

DEBUG: Failed to look up nuget package dotnetPackage
{
  "dependency": "dotnetPackage"
  "packageFile": "SimpleCalculator/SimpleCalculator.csproj"
}

```

What am I doing wrong? I have read all the things and tried in private-packae support:
[](https://github.com/renovatebot/renovate/blob/main/docs/usage/getting-started/private-packages.md)

Eventhough that document says this, I couldnt manage it.
![image](https://github.com/renovatebot/renovate/assets/1048693/c422ae59-058f-4fe3-a8ca-9f8351732973)

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by viceice** [comment]
`RestoreSources` are not supported by renovate. you should create a package rule matching nuget datasource and your depName. then add registry url

**Reply by viceice** [reply]
you can also configure a nuget.config

**Reply by CengAmat** [reply]
Thanks for answer @viceice 
Adding nuget.config works.
But I want to resolve it without adding nuget.config.

Removed `RestoreSources` from .csproj file.
and added matchDepName to the renovate.json, but still getting that warnign and same error in logs above.

What is wrong with renovate.json?

renovate.json:
```
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
        "config:recommended"
    ],
    "hostRules": [
     {
       "matchHost": "https://nuget.pkg.github.com/cengamat/",
       "hostType": "nuget",
       "encrypted": {
          "token": < PAT token that has all necessary read permissions  >
        }
     }
    ],
    "packageRules": [
     {
       "matchManagers": ["nuget"],
       "matchDepName": ["dotnetPackage"],
       "matchSourceUrls": ["https://nuget.pkg.github.com/cengamat/index.json"]
     }
   ]
}
```

**Reply by viceice** [reply]
registry url not match source url

**Reply by CengAmat** [reply]
Thank you very much mate.  Works now.

Is there a way to use secrets.github_token instead pat_token/password?

And I'm wondering why this didnt work before? Lack of nuget.config?
![image](https://github.com/renovatebot/renovate/assets/1048693/1952fcd3-f9ad-4ea9-a430-24120e34495a)

**Reply by viceice** [reply]
where did you found it? it was added and reverted because of issues.

the app token can't be used for packages

**Comment by CengAmat** [comment]
https://docs.renovatebot.com/getting-started/private-packages/#automatically-authenticate-for-npm-package-stored-in-private-github-npm-repository:~:text=For%20GitHub%20Packages,token%20by%20Renovate.

Thanks mate.
